identity-core/src/common/timestamp.rs

```diff
@@ -110,12 +123,40 @@ impl FromStr for Timestamp {
   }
 }
 
+/// Truncates an `OffsetDateTime` to the second.
+fn truncate_fractional_seconds(offset_date_time: OffsetDateTime) -> OffsetDateTime {
+  offset_date_time - Duration::nanoseconds(offset_date_time.nanosecond() as i64)
+}
+
 #[cfg(test)]
 mod tests {
+  use quickcheck::Arbitrary;
+  use quickcheck_macros::quickcheck;
+  const LAST_VALID_UNIX_TIMESTAMP: i64 = 253402300799;
+  const FIRST_VALID_UNIX_TIMESTAMP: i64 = -62167219200;
   use crate::common::Timestamp;
   use crate::convert::FromJson;
   use crate::convert::ToJson;
 
+  #[derive(Debug, Clone)]
+  struct ValidUnixTimestampFixture(i64);
+
+  impl Arbitrary for ValidUnixTimestampFixture {
+    fn arbitrary(g: &mut quickcheck::Gen) -> Self {
+      // collect 10_000 numbers between FIRST_VALID_UNIX_TIMESTAMP and LAST_VALID_UNIX_TIMESTAMP
+      // such that the distance between any two consecutive numbers is the same.
+      let interval_length = (LAST_VALID_UNIX_TIMESTAMP - FIRST_VALID_UNIX_TIMESTAMP) / 10_000;
+      let markers: Vec<i64> = (FIRST_VALID_UNIX_TIMESTAMP..LAST_VALID_UNIX_TIMESTAMP)
+        .step_by(interval_length as usize)
+        .collect();
+      let random_marker = *g.choose(&markers).unwrap();
+      // add another randomly chosen number between 1 and 1000 for extra randomness
+      let first_thousand: Vec<i64> = (1..=1000).collect();
+      let random_number = *g.choose(&first_thousand).unwrap();
+      Self(random_marker + random_number)
+    }
+  }
```

Comment:
Wow `quickcheck` is a lot of work. I would have just shoved it in a proptest range. You don't have to change this, I just prefer `proptest`.

E.g.
```rust
  proptest::proptest! {
    #[test]
    fn test_from_unix_to_rfc3339_valid_no_panic(seconds in FIRST_VALID_UNIX_TIMESTAMP..=LAST_VALID_UNIX_TIMESTAMP) {
      let timestamp = Timestamp::from_unix(seconds).unwrap();
      let expected_length = "dddd-dd-ddTdd:dd:ddZ".len();
      assert_eq!(timestamp.to_rfc3339().len(), expected_length);
    }

    #[test]
    fn test_from_unix_no_panic(seconds: i64) {
      let _ = Timestamp::from_unix(seconds);
    }
  }
```

Could also add a parse fuzz test then (in the same `proptest` block).

E.g.
```rust
    #[test]
    fn test_parse_no_panic(s in "\\PC*") {
      let _ = Timestamp::parse(s);
    }
```

Reply:
I also think `proptest` fits better here. I went with `quickcheck` because I saw that we were already using that in this crate, but since you would also prefer proptest, I will change this. 

>Edit2: doesn't the random selection exclude numbers that are not within 1000 of any of the intervals?

Yes it does. As far as I can tell one either has to allocate a vector with the entire range to make it available to `quickcheck`'s `choose` method, or bring in your own random generator.